spec/support/matchers/activity_log_matcher.rb

```diff
@@ -0,0 +1,41 @@
+# frozen_string_literal: true
+
+RSpec::Matchers.define :have_produced do |activity_type|
+  match do |actual|
+    @actual = actual
+    @activity_type = activity_type
+
+    expected_params = [@object, activity_type]
+    expected_params << {after_commit: @after_commit} unless @after_commit.nil?
+
+    expect(actual).to have_received(:produce).with(*expected_params)
+  end
+
+  chain :with do |object|
```

Comment:
Is `@object` always present? If yes, it should probably not be a chain as chains are used primarily for optional behaviors.

Reply:
It's probably required because I don't see how you can produce a log without an object, but I find this:

```rb
expect(Utils::ActivityLog).to have_produced("applied_coupon.deleted").with(applied_coupon)
```

so much nicer than this
```rbexpect(Utils::ActivityLog).to have_produced("applied_coupon.deleted").with(applied_coupon)
```